pcsx2/USB/usb-eyetoy/jo_mpeg.cpp

```diff
@@ -0,0 +1,471 @@
+/* public domain Simple, Minimalistic, No Allocations MPEG writer - http://jonolick.com
+ *
+ * Latest revisions:
+ * 	1.02 (22-03-2017) Fixed AC encoding bug.
+ *                    Fixed color space bug (thx r- lyeh!)
+ * 	1.01 (18-10-2016) warning fixes
+ * 	1.00 (25-09-2016) initial release
+ *
+ * Basic usage:
+ *	char *frame = new char[width*height*4]; // 4 component. RGBX format, where X is unused
+ *	FILE *fp = fopen("foo.mpg", "wb");
+ *	jo_write_mpeg(fp, frame, width, height, 60);  // frame 0
+ *	jo_write_mpeg(fp, frame, width, height, 60);  // frame 1
+ *	jo_write_mpeg(fp, frame, width, height, 60);  // frame 2
+ *	...
+ *	fclose(fp);
+ *
+ * Notes:
+ * 	Only supports 24, 25, 30, 50, or 60 fps
+ *
+ * 	I don't know if decoders support changing of fps, or dimensions for each frame.
+ * 	Movie players *should* support it as the spec allows it, but ...
+ *
+ * 	MPEG-1/2 currently has no active patents as far as I am aware.
+ * 	
+ *	http://dvd.sourceforge.net/dvdinfo/mpeghdrs.html
+ *	http://www.cs.cornell.edu/dali/api/mpegvideo-c.html
+ * */
+
+#include "PrecompiledHeader.h"
+#include <stdio.h>
+#include <math.h>
+#include <memory.h>
+
+#include "jo_mpeg.h"
+
+// Huffman tables
+static const unsigned char s_jo_HTDC_Y[9][2] = {{4, 3}, {0, 2}, {1, 2}, {5, 3}, {6, 3}, {14, 4}, {30, 5}, {62, 6}, {126, 7}};
+static const unsigned char s_jo_HTDC_C[9][2] = {{0, 2}, {1, 2}, {2, 2}, {6, 3}, {14, 4}, {30, 5}, {62, 6}, {126, 7}, {254, 8}};
+static const unsigned char s_jo_HTAC[32][40][2] = {
+	{
```

Comment:
can you please restore this file and jo_mpeg.h to the initial state ?

Reply:
Oh sorry, forgot about that! fixed